commands/audit/scarunner.go

```diff
@@ -107,38 +118,37 @@ func getRequestedDescriptors(params *AuditParams) map[coreutils.Technology][]str
 }
 
 // Preform the SCA scan for the given scan information.
-// This method will change the working directory to the scan's working directory.
-func executeScaScan(serverDetails *config.ServerDetails, params *AuditParams, scan *xrayutils.ScaScanResult) (err error) {
-	// Get the dependency tree for the technology in the working directory.
-	if err = os.Chdir(scan.WorkingDirectory); err != nil {
-		return errorutils.CheckError(err)
-	}
-	treeResult, techErr := GetTechDependencyTree(params.AuditBasicParams, scan.Technology)
-	if techErr != nil {
-		return fmt.Errorf("failed while building '%s' dependency tree:\n%s", scan.Technology, techErr.Error())
-	}
-	if treeResult.FlatTree == nil || len(treeResult.FlatTree.Nodes) == 0 {
-		return errorutils.CheckErrorf("no dependencies were found. Please try to build your project and re-run the audit command")
-	}
-	// Scan the dependency tree.
-	scanResults, xrayErr := runScaWithTech(scan.Technology, params, serverDetails, treeResult.FlatTree, treeResult.FullDepTrees)
-	if xrayErr != nil {
-		return fmt.Errorf("'%s' Xray dependency tree scan request failed:\n%s", scan.Technology, xrayErr.Error())
+func executeScaScan(auditParallelRunner *utils.AuditParallelRunner, serverDetails *config.ServerDetails, auditParams *AuditParams,
+	scan *xrayutils.ScaScanResult, flatTree xrayCmdUtils.GraphNode, fullDependencyTrees []*xrayCmdUtils.GraphNode) parallel.TaskFunc {
+	return func(threadId int) (err error) {
+		log.Info("[thread_id: "+strconv.Itoa(threadId)+"] Running SCA scan for", scan.Technology, "vulnerable dependencies in", scan.WorkingDirectory, "directory...")
+		defer func() {
+			auditParallelRunner.ScaScansWg.Done()
+		}()
+		// Scan the dependency tree.
+		xrayGraphScanParams := createXrayGraphScanParams(auditParams)
+		scanResults, xrayErr := runScaWithTech(scan.Technology, auditParams, xrayGraphScanParams, serverDetails, flatTree, fullDependencyTrees)
```

Comment:
```suggestion
		scanResults, xrayErr := runScaWithTech(scan.Technology, auditParams, serverDetails, flatTree, fullDependencyTrees)
```
This can be used and created inside `runScaWithTech` since we are passing it to the func, no need to add arg